### ISSUE TITLE
test_QApplication (kwallet): cannot connect to X server (nosetest with "env -")

### ISSUE THREAD

**Issue by jaraco** [jaraco/keyring#147]
I am the maintainer of the python-keyring package in Arch Linux (AUR) and I report a bug originally reported against the package:
https://aur.archlinux.org/packages/python-keyring/

I run `env - nosetest3 -v --with-id` during the package build process and when `kdebindings-python` is installed I get the following error:

```
#105 test_QApplication (keyring.tests.backends.test_kwallet.KDEKWalletInQApplication) ... : cannot connect to X server
```

The test already fails here:

```
#!python

>>> app = QApplication([])
: cannot connect to X server
```

Making DISPLAY and an X server available during the tests is not an option either. I tried it without `env -` and got lots of boxes opening asking me to create a keyring during tests. Unsuitable for tests running during packaging.

---
- Bitbucket: https://bitbucket.org/kang/python-keyring-lib/issue/147
- Originally reported by: Johannes Dewender
- Originally created at: 2014-05-06T14:28:40.185

**Comment by jaraco** [comment]
The original reporter tested with python-keyring 3.7 and kdebindings-python 4.13.0

I reproduced the problem with python-keyring 3.7 and kdebindings-python 4.11.4

---

Original comment by: Johannes Dewender

**Comment by jaraco** [comment]
I'm unfamiliar with the tests in the KDE environment.  Since it seems the tests require an active keyring to be configured, perhaps it makes sense to exclude those tests from your test run?

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
My change in pull request #46 fixes the problem for me. It skips the kwallet test if `DISPLAY` is not in the environment.

---

Original comment by: Johannes Dewender

**Comment by jaraco** [comment]
Fixed with pull request #46

---

Original comment by: Johannes Dewender
